backend/src/main/java/hu/flowacademy/konyhatunder/service/RecipeService.java

```diff
@@ -115,6 +114,27 @@ private String translateUnit(Measurement measurement, String unit) {
         return difficulties;
     }
 
+    public List<Recipe> sendRecipesByIngredients(List<Ingredient> ingredientList) {
+        validateReceivedIngredients(ingredientList);
+        List<Recipe> foundRecipes = new ArrayList<>();
+        for (Ingredient ingredient : ingredientList) {
+            List<Recipe> foundRecipeList = recipeRepository.findByIngredientsIngredientId(ingredient.getId());
+            for (Recipe foundRecipe : foundRecipeList) {
+                if (!foundRecipes.contains(foundRecipe)) {
```

Comment:
Itt a valtozo neveket irjuk at mert olvashatatlan. a foundRecipes az gyakorlatialg egyenlo a foundRecipeList-tel, nevezzuk at oket. 
Itt amugy is erdemes lesz egy normalis query-t irni ahelyett hogy backenden szurugetunk, de ha mar itt csinaljuk csinaljuk hatekonyabban:
Nincs szukseg a for loop-ra es a checkelesre, a jelenlegi foundRecipes legyen Set, aztan az addAll-al adjuk hozza a listat a set-hez ez gondoskodik rola hogy ne legyenek duplikaciok. Ebbol ha kell return elott tudunk List-et kesziteni.

Reply:
Done. Query folyamatban.